### ISSUE TITLE
how to use hardcoded `gasPrice` and `gasLimit` for all transactions

### ISSUE THREAD

**Issue by shunjizhan** [NomicFoundation/hardhat#2485]
Hi, according to the [doc](https://hardhat.org/hardhat-network/reference/#gas), in hardhat config, `gas` will override `gasLimit`, and `gasPrice` will override `gasPrice` for every transaction. However, when I set a hardcoded value for the gas params:
```
networks: {
  myNetwork: {
      gasPrice: 204369036266, 
      gas: 53064000,
  }
}
```
I saw my contract deployment transaction only used `204369036266` as `gasPrice`, but didn't use `53064000` as `gasLimit`, and it queried `eth_estimateGas` and used the result for `gasLimit`.

Is this expected? Did I misunderstand what the docs mean?

Basically I am trying to use these hardcoded gas params for **every** transaction, including contract deployment and call, in truffle if I put the same setting in truffle config it will work, so I was wondering how should I do it with hardhat? Thanks for the help!

**Comment by fvictorio** [comment]
Sadly this doesn't work with ethers.js. The docs should be clearer about this fact, sorry about that.

**Comment by fvictorio** [comment]
To clarify: this works when sending "low level" transactions, and I think it also works with `hardhat-web3`, but most people use ethers.js with `hardhat-ethers` now, so this part of the docs is misleading.

**Comment by shunjizhan** [comment]
Thanks! And it would be nice to clarify this in the doc so other devs won't get confused for this same issue : )

**Comment by fvictorio** [comment]
Oh yes, we have [an open PR](https://github.com/NomicFoundation/hardhat/pull/2406) for that, we just need to get it merged :slightly_smiling_face:
